native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2ConnectionManager.java

```diff
@@ -67,14 +80,27 @@ public void addHttp2ClientChannel(HttpRoute httpRoute, Http2ClientChannel http2C
      *
      * @param httpRoute  the route key
      */
-    public void releasePerRoutePoolLatch(HttpRoute httpRoute) {
-        String key = generateKey(httpRoute);
-        Http2ChannelPool.PerRouteConnectionPool perRouteConnectionPool = this.http2ChannelPool.fetchPerRoutePool(key);
+    public void releasePerRoutePoolLatch(EventLoop eventLoop, HttpRoute httpRoute) {
+        Http2ChannelPool.PerRouteConnectionPool perRouteConnectionPool = fetchPerRoutePool(httpRoute, eventLoop);
         if (perRouteConnectionPool != null) {
             perRouteConnectionPool.releaseCountdown();
         }
     }
 
+    /**
+     * Get or create the pool that is bound to the given eventloop.
+     *
+     * @param eventLoop netty event loop
+     * @return PerRouteConnectionPool
+     */
+    private Http2ChannelPool getOrCreateEventLoopPool(EventLoop eventLoop) {
+        final Http2ChannelPool pool = h2ChannelPools.get(eventLoop);
+        if (pool != null) {
+            return pool;
+        }
+        return h2ChannelPools.computeIfAbsent(eventLoop, e -> new Http2ChannelPool());
```

Comment:
I think following is enough cos it checks if the value associated to the key or else creates the value and insert it to the map.
```suggestion
        return h2ChannelPools.computeIfAbsent(eventLoop, e -> new Http2ChannelPool());
```